### PR TITLE
CH: Create basic CI pipeline

### PR DESCRIPTION
**Pull request by chrishylanduk**

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -62,12 +64,12 @@ async function registerForPushNotificationsAsync() {
...
-  const [expoPushToken, setExpoPushToken] = React.useState<String>("");
-  const [notification, setNotification] = React.useState<
-    Notifications.Notification | boolean
-  >(false);
-  const [userId, setUserId] = React.useState<String>("");
+function App(): ReactElement | null {
+  const [expoPushToken, setExpoPushToken] = React.useState<string>("");
+  // const [notification, setNotification] = React.useState<
+  //   Notifications.Notification | boolean
+  // >(false);
```
Comment (Aidan23Smith) [inline App.tsx:71]: Is this still needed?

File: App.tsx [hunk 2]
```diff
@@ -80,36 +82,28 @@ function App() {
...
 
     return () => {
-      if (notificationListener.current) {
+      if (notificationListener.current != null) {
         Notifications.removeNotificationSubscription(
           notificationListener.current
         );
       }
-      if (responseListener.current) {
+      if (responseListener.current != null) {
```
Comment (Aidan23Smith) [inline App.tsx:106]: I'm guessing the linter made you do this, but I did like it as it was

File: app/components/button/types.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
 export type buttonPropType = {
-  text: String;
+  text: string;
```
Comment (Aidan23Smith) [inline app/components/button/types.tsx:2]: Yes, I should stop putting a capital 'S'

File: app/components/logo_title/index.tsx [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import { Image } from "react-native";
 import * as React from "react";
+import { type ReactElement } from "react";
 
-function LogoTitle() {
+function LogoTitle(): ReactElement | null {
```
Comment (Aidan23Smith) [inline app/components/logo_title/index.tsx:5]: Can we remove null?

**Review by Aidan23Smith — APPROVED** [review]
A few comments, but happy with it all

File: App.tsx (hunk 1 above)
Comment (chrishylanduk) [reply App.tsx:71]: Lots of notification stuff will disappear when I rebase now

File: App.tsx (hunk 2 above)
Comment (chrishylanduk) [reply App.tsx:106]: Yeah linter wanted it. This one will disapper with rebase against OneSignal ticket, if it's an issue elsewhere then can discuss.

File: app/components/logo_title/index.tsx (hunk 4 above)
Comment (chrishylanduk) [reply app/components/logo_title/index.tsx:5]: I think so. For now have just made as generic as possible to get the linter to pass, as per https://stackoverflow.com/a/57363438

Comment (chrishylanduk) [reply app/components/logo_title/index.tsx]: Have changed all to just ReactElement

**Comment by chrishylanduk** [comment]
Did a quick manual test with the APK, seems fine

**Pull request merged**